tkg/client/update_cluster_metadata.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"github.com/pkg/errors"
+	yaml "gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+	clusterctlv1 "sigs.k8s.io/cluster-api/cmd/clusterctl/api/v1alpha3"
+	crtclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/tanzu-framework/tkg/clusterclient"
+	"github.com/vmware-tanzu/tanzu-framework/tkg/constants"
+)
+
+const clusterMetadataNamespace = "tkg-system-public"
+const tkgBomConfigMapName = "tkg-bom"
+const tkgMetadataConfigMapName = "tkg-metadata"
+
+// CreateOrUpdateClusterMetadata will create or update the tkg-bom ConfigMap and tkg-metadata ConfigMap
+func (c *TkgClient) CreateOrUpdateClusterMetadata(regionalClusterClient, currentClusterClient clusterclient.Client, isRegionalCluster bool, edition string, clusterName string, clusterNamespace string) error {
+	type ConfigmapRef struct {
+		Name string `yaml:"name"`
+	}
+
+	type Bom struct {
+		ConfigmapRef *ConfigmapRef `yaml:"configmapRef"`
+	}
+
+	type Infrastructure struct {
+		Provider string `yaml:"provider"`
+	}
+
+	type Cluster struct {
+		Name               string          `yaml:"name"`
+		Type               string          `yaml:"type"`
+		Plan               string          `yaml:"plan"`
```

Comment:
Could we also add metadata on whether this is a classy cluster or legacy cluster. I am sure this will be useful to determine certain behaviors for users.

Reply:
Thanks @shyaamsn ! I added the isClusterClassBased field to record that. Please take a look